### ISSUE TITLE
Necessity of --sessionid argument

### ISSUE THREAD

**Issue by yoursunny** [justincpresley/ndn-hydra#17]
Currently, `ndn-hydra-repo` command requires a `--sessionid` argument, which must be unique when the repo is restarted.
Having it as a required argument is error-prone because the user may inadvertently enter a duplicate value.
(I often use bash `$RANDOM` but it may have collisions.)

It is safer and more user friendly to delete this argument and generate a random number as sessionid.
For example, the random sessionid could have 8 octets consisting of:

* 32-bit timestamp (microseconds since epoch truncated to lower 32 bits)
* 32-bit cryptographical secure random number

**Comment by justincpresley** [comment]
Background
-------------
We originally planned for the `node_name`  to be used for everything and not have a `sessionid` at all. These two things changed our mind:
* SVS does not allow you to remove any IDs that are put within it even if they go offline. This creates a problem of discerning what nodes are actually available or not.
* The current implementation of SVS (which is updated to the latest spec) has a ID limit (# of nodes that can join) that is directly proportional to the sizes of the IDs that are used.

Design
--------
Unless you are in the group using an ID, you are unable to use what ID have been used or are currently in use. We decided for general use right now that a sessionid of 4-6 characters including numbers i.e. "e111" was the best for SVS IDs. Smaller values could limit long-term use and expandability, larger values would limit expandability.

Our options for generating / choosing a `sessionid` of that length:
* have a file that keeps track of the last session id used, generate a new one specific to the node using a certain method
* contact another node to get all the `sessionid` info and use an ID that hasn't been taken
* have the user define it and keep track of the sessions
* generate a random `sessionid` of the desired characteristics

We did choose the third option for now.  If its alright with you, we can explore this problem further after we incorporate "hydra logging" which will log file fetching, all files history, restarts and errors, etc.

I do agree and see your point. It is definitely safer to hide this from the user.

If you have any further suggestion(s) associated to this problem after considering the above, I would love to hear it.
